src/goto-instrument/loop_utils.cpp

```diff
@@ -53,14 +53,60 @@ void get_assigns_lhs(
   assignst &assigns,
   std::function<bool(const exprt &)> predicate)
 {
-  if(
-    (lhs.id() == ID_symbol || lhs.id() == ID_member || lhs.id() == ID_index) &&
-    predicate(lhs))
+  assignst new_assigns;
+
+  if((lhs.id() == ID_symbol || lhs.id() == ID_index))
+  {
+    // All variables `v` and their indexing expressions `v[idx]` are assigns.
+    new_assigns.insert(lhs);
+  }
+  else if(lhs.id() == ID_member)
   {
-    assigns.insert(lhs);
+    auto op = to_member_expr(lhs).struct_op();
+    auto component_name = to_member_expr(lhs).get_component_name();
+
+    // Insert expressions of form `v.member`.
+    if(op.id() == ID_symbol)
+    {
+      new_assigns.insert(lhs);
+    }
+
+    // For expressions of form `v->member`, insert all targets `u->member`,
+    // where `u` and `v` alias.
+    else if(op.id() == ID_dereference)
+    {
+      const pointer_arithmetict ptr(to_dereference_expr(op).pointer());
+      for(const auto &mod : local_may_alias.get(t, ptr.pointer))
+      {
+        const typecast_exprt typed_mod{mod, ptr.pointer.type()};
```

Comment:
Nit pick: move this after the `if(mod.id() == ID_unknown)` for there is no need to construct such an object if that `if` condition is true.

Reply:
Also, use `typecast_exprt::conditional_cast` as that typecast might not actually be necessary.